src/components/Markdown/Markdown.tsx

```diff
@@ -31,15 +44,185 @@ export type MarkdownProps = Pick<
+
+  const handleShow = useCallback(() => {
+    if (typeof children !== "string") return;
+    if (!markdown) return;
+
+    getPatch({ pin: children, markdown })
+      .unwrap()
+      .then((patch) => {
+        if (patch.chunks.length === 0) {
+          throw new Error("No Chunks to show");
+        }
+        diffPreview(patch);
+      })
+      .catch(() => {
+        setErrorMessage({ type: "warning", text: "No patch to show" });
+      });
+  }, [children, diffPreview, getPatch, markdown]);
+
+  const handleApply = useCallback(() => {
+    if (typeof children !== "string") return;
+    if (!markdown) return;
+    getPatch({ pin: children, markdown })
+      .unwrap()
+      .then((patch) => {
+        const files = patch.results.reduce<string[]>((acc, cur) => {
+          const { file_name_add, file_name_delete, file_name_edit } = cur;
+          if (file_name_add) acc.push(file_name_add);
+          if (file_name_delete) acc.push(file_name_delete);
+          if (file_name_edit) acc.push(file_name_edit);
+          return acc;
+        }, []);
+
+        if (files.length === 0) {
+          setErrorMessage({ type: "warning", text: "No chunks to apply" });
+          return;
+        }
+
+        const fileIsOpen = files.some((file) => openFiles.includes(file));
+
+        if (fileIsOpen) {
+          diffPreview(patch);
+        } else {
+          const chunks = patch.chunks;
+          const toApply = chunks.map(() => true);
+          void onSubmit({ chunks, toApply });
+        }
+      })
+      .catch((error: Error) => {
+        setErrorMessage({
+          type: "error",
+          text: error.message
+            ? "Failed to apply patch: " + error.message
+            : "Failed to apply patch.",
+        });
+      });
+  }, [children, diffPreview, getPatch, markdown, onSubmit, openFiles]);
+
+  useEffect(() => {
+    if (errorMessage) {
+      setTimeout(() => setErrorMessage(null), 3000);
+    }
+  }, [errorMessage]);
+
+  if (isPin) {
+    return (
+      <Box>
+        <Flex my="2" gap="2" wrap="wrap-reverse">
+          <Text
+            as="p"
+            wrap="wrap"
+            style={{ lineBreak: "anywhere", wordBreak: "break-all" }}
+          >
+            {children}
+          </Text>
+          <Flex gap="2" justify="end" ml="auto">
+            {host !== "web" && (
+              <Button
+                size="1"
+                // loading={patchResult.isFetching}
+                onClick={handleShow}
+                title="Show Patch"
+                disabled={!!errorMessage}
+              >
+                Open
+              </Button>
+            )}
+            <Button
+              size="1"
+              // loading={patchResult.isFetching}
+              onClick={handleApply}
+              disabled={!!errorMessage}
+              title="Apply patch"
+            >
+              Apply
+            </Button>
+          </Flex>
+        </Flex>
+        {errorMessage && (
```

Comment:
Why did you not use `ErrorCallout` component? I agree that error state from redux is not appliable here, but the component by itself could be used here instead of creating a new callout.

Reply:
Done :)